extensions/resource-deployment/src/services/azdataService.ts

```diff
@@ -0,0 +1,134 @@
+				if (error) {
+					reject(error.message);
+				} else {
+					try {
+						const output = <BdcConfigListOutput>JSON.parse(stdout);
+						resolve(output.stdout);
+					}
+					catch (err) {
+						if (err instanceof Error) {
+							reject(err.message);
+						} else {
+							reject(err);
+						}
+					}
+				}
+			});
+		});
+		return promise;
+	}
+
+	private getDeploymentProfileInfo(profileName: string): Thenable<DeploymentProfile> {
+		const promise = new Promise<DeploymentProfile>((resolve, reject) => {
+			if (!fs.existsSync(this.platformService.storagePath())) {
+				fs.mkdirSync(this.platformService.storagePath());
+			}
+			cp.exec(`azdata bdc config init --source ${profileName} --target ${profileName} --force`, { cwd: this.platformService.storagePath() }, (error, stdout, stderror) => {
+				if (error) {
+					reject(error.message);
+				} else {
+					try {
+						const bdcJson = this.getJsonObject(path.join(this.platformService.storagePath(), profileName, 'bdc.json'));
+						const controlJson = this.getJsonObject(path.join(this.platformService.storagePath(), profileName, 'control.json'));
+						resolve({
+							name: profileName,
+							defaultDataSize: (<string>controlJson.spec.storage.data.size).replace('Gi', ''),
+							defaultLogSize: (<string>controlJson.spec.storage.logs.size).replace('Gi', ''),
+							defaultDataStorageClass: (<string>controlJson.spec.storage.data.className),
+							defaultLogsStorageClass: (<string>controlJson.spec.storage.logs.className),
+							master: <string>bdcJson.spec.resources['master'].spec.replicas,
+							data: <string>bdcJson.spec.resources['data-0'].spec.replicas,
+							compute: <string>bdcJson.spec.resources['compute-0'].spec.replicas,
+							hdfs: <string>bdcJson.spec.resources['storage-0'].spec.replicas,
+							nameNode: <string>bdcJson.spec.resources['nmnode-0'].spec.replicas,
+							spark: '0',
+							activeDirectory: false, // TODO: implement AD check
+							hadr: <string>bdcJson.spec.resources.master.spec.settings.sql['hadr.enabled'],
+							includeSpark: <string>bdcJson.spec.resources['storage-0'].spec.settings.spark.includeSpark,
+							gatewayPort: this.getEndpointPort(bdcJson.spec.resources.gateway.spec.endpoints, 'Knox'),
+							appProxyPort: this.getEndpointPort(bdcJson.spec.resources.appproxy.spec.endpoints, 'AppServiceProxy'),
+							sqlServerPort: this.getEndpointPort(bdcJson.spec.resources.master.spec.endpoints, 'Master'),
+							readableSecondaryPort: this.getEndpointPort(bdcJson.spec.resources.master.spec.endpoints, 'MasterSecondary', '31436'),
+							controllerPort: this.getEndpointPort(controlJson.spec.endpoints, 'Controller')
+						});
+					}
+					catch (err) {
+						if (err instanceof Error) {
+							reject(err.message);
+						} else {
+							reject(err);
+						}
+					}
+				}
+			});
+		});
+		return promise;
+	}
+
+	private getJsonObject(path: string): any {
+		return JSON.parse(fs.readFileSync(path, 'utf8'));
+	}
```

Comment:
Don't use sync methods. Make this async and using the fs.promises versions (i.e. fs.promises.readFile)

Reply:
looks like you are not looking at the latest code, this has been changed.

Reply:
Yeah, sorry that was out of date. See other comment I just left for a suggestion on making this much cleaner

---
In reply to: [326371213](https://github.com/microsoft/azuredatastudio/pull/7183#discussion_r326371213) [](ancestors = 326371213)